README.md

```diff
@@ -1,151 +1,227 @@
+```sh
+$ npx elm-translations -f en.json -o src
+```
 
+2. In your Html page pass the translation data:
 
-parse : Value -> Result Error Translations
-parse =
-    Json.Decode.decodeValue decodeTranslations
+```html
+...
+<script>
+  Elm.Main.init({
+    node: document.getElementById("elm"),
+    flags: {
+      // your translation data inlined by e.g. EJS or Handlebars:
+      // ...
+      welcome: "Welcome {{name}}!",
+      home: {
+        intro: "This App is about ..."
+      }
+      // ...
+    }
+  });
+</script>
+...
 ```
 
-### Use it in your Elm app like e.g.:
+3. In your Elm app use the translations:
 
 ```elm
-module Main exposing (main)
+module Main exposing (..)
 
-import Browser
-import Html exposing (..)
 import Json.Decode
-import Translations
+import Json.Encode
+import Translations exposing (Translations)
 
 
-main : Program Flags Model Msg
-main =
-    Browser.element
-        { init = init
-        , view = view
-        , update = update
-        , subscriptions = subscriptions
-        }
+-- ...
 
 
 type Model
-    = Failed String
-    | Initialised Translations.Translations
+    = Initialized Translations
+    | Failed Json.Decode.Error
 
 
-type alias Flags =
-    { translations : Json.Decode.Value }
-
-
-init : Flags -> ( Model, Cmd Msg )
-init { translations } =
-    case Translations.parse translations of
+init : Json.Encode.Value -> ( Model, Cmd Msg )
+init flags =
+    case Translations.parse flags of
         Ok t ->
-            ( Initialised t
-            , Cmd.none
-            )
+            ( Initialized t, Cmd.none )
 
-        Err _ ->
-            ( Failed "Parsing translations failed"
-            , Cmd.none
-            )
+        Err error ->
+            ( Failed error, Cmd.none )
 
 
 view : Model -> Html Msg
 view model =
     case model of
-        Failed message ->
-            Html.text <| "Initialisation failed: " ++ message
-
-        Initialised t ->
-            p []
-                [ Html.text t.keywords
-                , br [] []
-                , Html.text <|
-                    t.navigation.admin.authInfo
-                        { username = "John D."
-                        , role = "Admin"
-                        }
+        Initialized t ->
+            div []
+                [ h1 [] [ text <| t.welcome { name = "John" } ]
+                , p [] [ text t.home.intro ]
                 ]
 
-...
+        Failed error ->
+            p [] [ text "Error: Cannot proccess translation data" ]
+
+
+-- ...
+
+```
+
+### Dynamically fetching translation data
+
+There's an example available in the git repository. To let it run locally on your machine, follow these steps:
+
+1. Clone the respository
+
+```sh
+$ git clone git@github.com:layflags/elm-translations.git
+```
+
+2. Go to the example folder
+
+```sh
+$ cd elm-translations/example
+```
+
+3. Generate the `Translations.elm` module
+
+```sh
+$ npx elm-translations --from en.json --out src
```

Comment:
This is not deployed yet, so pls do `../index.js --from en.json --out src` while testing